File: Java/singleNumber.java
// O(N^2)
class Solution {
    public int singleNumber(int[] nums) {
        for(int i=0; i<nums.length; i++){
            int k= nums[i];
            boolean bool= false;

            for(int j=0; j<nums.length; j++){
                if(i!=j){
                    int x= k^nums[j];
                    if(x==0){
                        bool= true;
                        break;
                    }
                }
            }

            if(!bool){
                return k;
            }
        }

        return -1;
    }
}

// O(N)

class Solution {
    public int singleNumber(int[] nums) {
        int ans=0;
        for(int i=0;i<=nums.length-1;i++){
            ans^=nums[i];
        }
        return ans;
    }
}

File: Python/singleNumber.py
# O(N^2)
class Solution:
    def singleNumber(self, nums: list[int]) -> int:
        for i in range(len(nums)):
            k = nums[i]
            boolValue = False

            for j in range(len(nums)):
                if i != j:
                    x = k ^ nums[j]
                    if x == 0:
                        boolValue = True
                        break

            if not boolValue:
                return k

        return -1

# O(N)
class Solution:
    def singleNumber(self, nums: list[int]) -> int:
        ans = 0
        for num in nums:
            ans ^= num
        return ans

